.clang-tidy

```diff
@@ -12,6 +12,7 @@ Checks:  '-*,
           -cppcoreguidelines-pro-type-vararg,
           -cppcoreguidelines-pro-type-union-access,
           -cppcoreguidelines-non-private-member-variables-in-classes,
+          -cppcoreguidelines-owning-memory,
```

Comment:
added because I personally don't think we need to enable the `gsl` cpp core guidelines stuff in our repo. maybe could revisit it at a later date, but it adds another barrier to entry for new members.